### PR TITLE
Promisify 'autoupdate'

### PR DESCRIPTION
**Pull request by PradnyaBaviskar**
Connect https://github.com/strongloop/loopback/issues/418

@bajtos,
Please review.

### CODE REVIEW

File: test/memory.test.js [hunk 1]
```diff
@@ -548,6 +548,42 @@ describe('Memory connector', function() {
...
+      ds.autoupdate()
+        .then(function(result) {
+            done();
+          })
+        .catch(function(err){
+          done(err);
+        });
+    });
+  });
+
```
Comment (PradnyaBaviskar) [inline test/memory.test.js]: I was planning to add test cases for `autoupdate`, using -

```
ds = new DataSource({
  connector: 'memory'
});
```
But then realized that memory connector does not have `autoupdate` method. Hence skipped both the test cases above. How should I be verifying `promise` support for `autoupdate`?

Comment (PradnyaBaviskar) [inline test/memory.test.js]: Oh got it, in `beforeEach`, I can add a mock `autoupdate`. Something like this -

```
ds.autoupdate = function(models, cb) {
  process.nextTick(cb);
};
```

Will update my test suite, for this.

Comment (bajtos) [inline test/memory.test.js]: :+1:

**Comment by PradnyaBaviskar** [comment]
Done with adding test cases for autoupdate.
Can you please review?

**Comment by bajtos** [comment]
@slnode test please

**Comment by bajtos** [comment]
Landed, thank you!

**Pull request merged**